### PR TITLE
Pause and resume methods and autostop option

### PR DESCRIPTION
**Pull request by Stending**
## Description
Added a pause and resume method for Vuforia. They'll just stop and start the trackers, so the view will still show the camera, but you'll not be able to detect a target.
It can be really useful if you want some phases where you don't always want the player to scan a target.

Also, I made optional turning off vuforia when an image is found. It's true by default, so it'll change nothing, but if you add `autostopOnImageFound: false` in the options object, it'll just stop trackers when an image is found, and you'll have to stop vuforia manually (with StopVuforia).

## Motivation and Context
After start and stop vuforia methods, a pause and a resume methods where interesting.
And the non-autostop on image found is really useful for us.

## How Has This Been Tested?
Tested on iPod Touch and Galaxy S2.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by mattrayner** [comment]
I have to question the true usefulness of this, can you give me a concrete use case where, as part of a Cordova application this is needed? Also, can you squash your commits?

**Comment by mattrayner** [comment]
I also have concerns about some of the code you're submitting at the moment - there seems to be a lot of repetition, and your pull request isn't clean. it needs to be rebased off of master, not just merged in.

This pull request is currently at 21 commits but includes a lot of code that already exists in the code base.

A rebase instead of merge will shorten the commits to just the changes you've made. In it's current state I wouldn't be happy to merge this code into the plugin.

**Comment by Stending** [comment]
Wow, sorry for that, I did a lot of stupid things.
Everything should be great now.

Here're some examples of case that can use those features:

In our app, we need to play a sound when we find a target. If we play the sound in the callback method (like you do an alert in your example), the sound is offset by 0.5 second. So we need to just stop the trackers, play the sound in the callback method, and then stop vuforia manually.

So that's in our case, but now imagine you want an app that ask you to scan 10 images, but you don't want to stop vuforia after each image. Then you just have to set the autostopOnImageFound to false, and when you find an image, you increment the number of images found, and after X ms, you resume the trackers with resumeVuforia(). When it exceed 10, you stop vuforia and continue your app.

Currently, the plugin stops the trackers automatically when an image is found, so you don't have to do pauseVuforia() for that.

File: src/android/VuforiaPlugin.java [hunk 1]
```diff
@@ -114,6 +118,26 @@ else if(action.equals("cordovaStopVuforia")) {
...
+
+            JSONObject json = new JSONObject();
+            json.put("success", "true");
+            callback.sendPluginResult(new PluginResult(PluginResult.Status.OK, json));
+
+            Intent pauseIntent = new Intent(PLUGIN_ACTION);
+            pauseIntent.putExtra(PLUGIN_ACTION, PAUSE_ACTION);
+            this.cordova.getActivity().sendBroadcast(pauseIntent);
+        }else if(action.equals("resumeVuforia")){
+            Log.d(LOGTAG, "Resuming trackers");
```
Comment (mattrayner) [inline src/android/VuforiaPlugin.java]: There's a lot of duplication here, can you refactor so your code is DRY?

**Comment by Stending** [comment]
Done!

**Comment by mattrayner** [comment]
Cool, can you rebase this branch off the latest master?

**Comment by Stending** [comment]
I did other stupid things, but now everything is fine, and I rebased it off the master!

**Comment by mattrayner** [comment]
Requires additional manual merging - see #71

**Pull request closed without merging**